components/earnings/commons/earnings_base.lua

```diff
@@ -0,0 +1,185 @@
+local Class = require('Module:Class')
+
+local _DEFAULT_DATE = '1970-01-01 00:00:00'
+local _FIRST_DAY_OF_YEAR = '-01-01'
+local _LAST_DAY_OF_YEAR = '-12-31'
+
+---
+-- Entry point for players and individuals
+-- @player - the player/individual for whom the earnings shall be calculated
+-- @year - (optional) the year to calculate earnings for
+-- @mode - (optional) the mode to calculate earnings for
+-- @noRedirect - (optional) player redirects get not resolved before query
+-- @prefix - (optional) the prefix under which the players are stored in the placements
+-- @playerNumber - (optional) the number for how many params the query should look in LPDB
+-- @startYear - (optional) query yearly earning starting with that year and return the values in a lua table
+function Earnings.calculateForPlayer(args)
+	args = args or {}
+	local player = args.player
+
+	if String.isEmpty(player) then
+		return 0
+	end
+	if not Logic.readBool(args.noRedirect) then
+		player = mw.ext.TeamLiquidIntegration.resolve_redirect(player)
+	end
+
+	local prefix = args.prefix or 'p'
+
+	local playerNumber = tonumber(args.playerNumber) or 10
+	if playerNumber <=0 then
+		error('"playerNumber" has to be >= 1')
+	end
+
+	local playerConditions = '([[participant::' .. player .. ']]'
+	for playerIndex = 1, playerNumber do
+		playerConditions = playerConditions .. ' OR [[players_' .. prefix .. playerIndex .. '::' .. player .. ']]'
+	end
+	playerConditions = playerConditions .. ')'
+
+	return Earnings._calculate(playerConditions, args.year, args.mode, args.startYear, Earnings.divisionFactor)
+end
+
+---
+-- Entry point for teams
+-- @team - the team for which the earnings shall be calculated
+-- @year - (optional) the year to calculate earnings for
+-- @mode - (optional) the mode to calculate earnings for
+-- @noRedirect - (optional) player redirects get not resolved before query
+-- @startYear - (optional) query yearly earning starting with that year and return the values in a lua table
+function Earnings.calculateForTeam(args)
+	args = args or {}
+	local team = args.team
+
+	if String.isEmpty(team) then
+		return 0
+	end
+	if not Logic.readBool(args.noRedirect) then
+		team = mw.ext.TeamLiquidIntegration.resolve_redirect(team)
+	end
+
+	local teamConditions = '([[participant::' .. team .. ']] OR [[extradata_participantteam::' .. team .. ']])'
+
+	return Earnings._calculate(teamConditions, args.year, args.mode, args.startYear, Earnings._divisionFactorOne)
+end
+
+---
+-- Calculates earnings for this participant in a certain mode
+-- @participantCondition - the condition to find the player/team
+-- @year - (optional) the year to calculate earnings for
+-- @mode - (optional) the mode to calculate earnings for
+function Earnings._calculate(conditions, year, mode, startYear, divisionFactor)
+	conditions = Earnings._buildConditions(conditions, year, mode, startYear)
+
+	if String.isNotEmpty(startYear) then
+		return Earnings._calculatePerYear(conditions, divisionFactor)
+	end
```

Comment:
This would we weird no? 
If you enter any startYear, then you receive a table for all years as output?
Also startYear parameter isn't used anywhere except in this if statement.

Reply:
I see this was resolved while I was writing comments.